InventoryIndexer/Model/IsProductSalable.php

```diff
@@ -0,0 +1,48 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryIndexer\Model;
+
+use Magento\Framework\Exception\LocalizedException;
+use Magento\InventorySalesApi\Api\IsProductSalableInterface;
+use Magento\InventorySalesApi\Model\GetStockItemDataInterface;
+
+/**
+ * @inheritDoc
+ *
+ * Lightweight implementation for Storefront application.
+ */
+class IsProductSalable implements IsProductSalableInterface
+{
+    /**
+     * @var GetStockItemDataInterface
+     */
+    private $stockItemData;
+
+    /**
+     * @param GetStockItemDataInterface $stockItemData
+     */
+    public function __construct(GetStockItemDataInterface $stockItemData)
+    {
+        $this->stockItemData = $stockItemData;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function execute(string $sku, int $stockId): bool
+    {
+        try {
+            $stockItem = $this->stockItemData->execute($sku, $stockId);
+            $isSalable = (bool)($stockItem[GetStockItemDataInterface::IS_SALABLE] ?? false);
```

Comment:
```suggestion
            $isSalable = (bool)($stockItem[GetStockItemDataInterface::IS_SALABLE]);
```

Reply:
`$stockItem` can be `null`.

Reply:
`null` converted to bool is `false`

Reply:
`null['somevalue']` is `null`

Reply:
![image](https://user-images.githubusercontent.com/26086561/81428561-10ee5280-9165-11ea-83df-853e902ce348.png)

Reply:
<img width="1044" alt="Screen Shot 2020-05-09 at 9 25 10 PM" src="https://user-images.githubusercontent.com/9420807/81481858-a825d980-923b-11ea-967b-9babef0943ac.png">
Ok, but we will have troubles in future.